### PR TITLE
test: Don't read qemu console output during boot.

### PR DESCRIPTION
**Pull request by mvollmer**
This isn't reliable anymore: https://bugzilla.redhat.com/show_bug.cgi?id=1205529

To compensate, we import the "mac to ip" mapping from the network
definition and wait for "/run/nologin" to disappear to signal that
user logins are now possible.

### CODE REVIEW

**Comment by mvollmer** [comment]
This is an alternative to #2035.

File: test/testvm.py [hunk 1]
```diff
@@ -36,6 +36,33 @@
...
+    '52:54:00:9e:00:0c': '10.111.111.22',
+    '52:54:00:9e:00:0d': '10.111.111.23',
+    '52:54:00:9e:00:0e': '10.111.111.24',
+    '52:54:00:9e:00:0f': '10.111.111.25',
+
+    '52:54:00:9e:00:F0': '10.111.111.100',
+    '52:54:00:9e:00:F1': '10.111.111.101',
+    '52:54:00:9e:00:F2': '10.111.111.102',
+    '52:54:00:9e:00:F3': '10.111.111.103',
+    '52:54:00:9e:00:F4': '10.111.111.104'
```
Comment (stefwalter) [inline test/testvm.py]: I suggest we don't duplicate this information. Ideally we would ask dnsmasq about it. But if we can't we should at least keep it in one place.

File: test/testvm.py [hunk 2]
```diff
@@ -98,6 +125,19 @@ def wait_ssh(self):
...
+        tries_left = 5
+        while (tries_left > 0):
+            try:
+                self.execute("! test -f /run/nologin")
+                return
+            except:
+                pass
+            tries_left = tries_left - 1
+            time.sleep(1)
+        raise Failure("Timed out waiting for /run/nologin to disappear")
```
Comment (stefwalter) [inline test/testvm.py]: Why don't we use @dperpeet's patch to connect to the socket here?

**Comment by stefwalter** [comment]
Fails during ./vm-install stage of VERIFY like this:

```
Traceback (most recent call last):
  File "./vm-install", line 32, in <module>
    machine.install(args.rpms)
  File "/data/src/cockpit/test/testvm.py", line 218, in install
    self.stop()
  File "/data/src/cockpit/test/testvm.py", line 725, in stop
    self.shutdown()
  File "/data/src/cockpit/test/testvm.py", line 787, in shutdown
    self.wait_poweroff()
  File "/data/src/cockpit/test/testvm.py", line 760, in wait_poweroff
    fd = self._process.stdout.fileno()
AttributeError: 'NoneType' object has no attribute 'fileno'
```

File: test/testvm.py (hunk 2 above)
Comment (mvollmer) [inline test/testvm.py]: We do both.  Ssh starts working for root before it starts working for non-root.  See 1bb4ba35.

Comment (stefwalter) [inline test/testvm.py]: Aha. Very good. How about adding a comment explaining that?

**Comment by mvollmer** [comment]
> Fails during ./vm-install stage of VERIFY like this:

Fixed.

File: test/testvm.py (hunk 1 above)
Comment (mvollmer) [inline test/testvm.py]: Yes, I agree.

File: test/testvm.py [hunk 3]
```diff
@@ -671,55 +709,13 @@ def start(self, maintain=False):
...
-                sys.stdout.write("Boot log:\n" + all_output + "\n")
-                raise Failure("qemu vm boot timed out")
-
-        if p.poll():
-            raise Failure("qemu did not run successfully: %d" % (p.returncode, ))
 
-        self.address = address
+        self.address = ip_from_mac[self.macaddr]
         if not Machine.wait_ssh(self):
             raise Failure("Unable to reach machine %s via ssh." % (address))
```
Comment (dperpeet) [inline test/testvm.py]: this needs to be self.address now

File: test/testvm.py (hunk 1 above)
Comment (mvollmer) [inline test/testvm.py]: Fixed by putting the XML into its own file.

File: test/testvm.py (hunk 2 above)
Comment (mvollmer) [inline test/testvm.py]: Added.

**Comment by stefwalter** [comment]
This is much cooler than #2035. I've still merged #2035, with the thought that this would replace it. Given the state of hubbot on this commit, it seems like it'll need a bit of work before it's ready.

**Comment by mvollmer** [comment]
> Given the state of hubbot on this commit, it seems like it'll need a bit of work before it's ready.

Yes.  @dperpeet has made #2069 with a lot of improvements.

**Comment by mvollmer** [comment]
I have treated #2069 as a review comment and included the essential fixes here.

File: test/testvm.py [hunk 4]
```diff
@@ -615,9 +633,8 @@ def _start_qemu(self, maintain=False, tty=False, monitor=None):
             return subprocess.Popen(cmd)
 
         quoted = " ".join("'%s'" % x for x in cmd)
-        cmd = [ "/bin/sh", "-c", quoted + " | tee " + self.run_dir + "/qemu-$$.log" ]
-
-        return subprocess.Popen(cmd, stdout=subprocess.PIPE, stdin=subprocess.PIPE)
+        cmd = [ "/bin/sh", "-c", quoted + " </dev/null >/dev/null" ]
```
Comment (mvollmer) [inline test/testvm.py]: This could be improved.  We don't need the shell, and we could redirect qemu output directly to a file.

Comment (mvollmer) [inline test/testvm.py]: Done.

**Pull request closed without merging**